### ISSUE TITLE
Some properties named typeParameters instead of typeArguments

### ISSUE THREAD

**Issue by dsherret** [typescript-eslint/typescript-eslint#146]
I've noticed several parts of the code referring to type arguments as type parameters. What's the reason for using `typeParameters` here? Seems like there might be some temporary stuff going on? Will this be renamed in the future?

```ts
if (node.typeArguments && node.typeArguments.length) {
    result.typeParameters = convertTypeArgumentsToTypeParameters(
        node.typeArguments
    );
}
```

https://github.com/typescript-eslint/typescript-eslint/blob/45d8ad3a9422365a19d29c1452a667aa2b8a1049/packages/typescript-estree/src/convert.ts#L2045-L2050

Some example code: [playground](https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&showAST=es&code=M4ewtgpgYgrgdgYwC4EsRwDzCQJxXAcwD4AKASgG4ACAehqqQAsVgGBPABwioEMcCYkOEiotRiEDhwRkAGzZVpAMwhSIAEwYherHuy5UOfHpCSrxDRtx7YAsAChQkWIlToSARkq16TMXwEhETEESWk5BWVVaU0kbRteOF5+QQhhCyZrbCA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZYgQzMVoCNEAMwD20XoNhMy+SsKa0uYpujABtcNhyJo0UZAA0GzVk3ZInJnMIBbYbFQZz8AO6dCqQ6bNNO1xABMVSCZEADdtAyNTc2RCKQBBHWFnIM4XN1QoiABfKIBdDVzsoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)

```ts
someFunction<string>(); // this type argument is incorrectly referred to as a type parameter in the ast
someFunction(1); // this argument is correctly referred to as an argument in the ast
```

Anyway, I just wanted to start a discussion about it so at least it's in the issue tracker. Also, FWIW, babel-eslint9 correctly refers to them as `typeArguments` when I'm looking at https://astexplorer.net, but I'm not so familiar with the babel or eslint ecosystem so I'm not sure how they're related (Edit: Looks like they call it `node.typeArguments.params` though...).

**Comment by dsherret** [comment]
@armano2 sorry, I think my bad (first day looking at all this code). This looks like a question I should be raising in babel-parser? See here:

https://github.com/babel/babel/blob/03230eaa9ce198bdd9dcff048ab78f52d85ec55b/packages/babel-parser/src/plugins/typescript.js#L1532

**Comment by dsherret** [comment]
I opened https://github.com/babel/babel/issues/9418 ...I'm not sure how this project relates to babel's parser (haven't looked into it much).

**Comment by armano2** [comment]
we are trying to make AST ~same as babel to improve user experience, but we have some changes to AST
(they have broken stuff), i'm regularly reporting issues there (and time to time fixing them)

you can find know issues with babel AST here: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/typescript-estree/tests/ast-alignment/utils.ts#L84

**Comment by armano2** [comment]
note: investigate before 6.0.0 release

**Comment by JoshuaKGoldberg** [comment]
I threw up a draft of what this could look like on our end: https://github.com/typescript-eslint/typescript-eslint/pull/5384

Unless I've misinterpreted the code, it looks like we'll need Babel to also apply these suggested changes (https://github.com/babel/babel/issues/9418) first. Marking as blocked until then.

**Comment by bradzacher** [comment]
We don't need babel to maintain compat with us. It's nice if they do but there's no requirement.

We already have a number of divergences in the AST which makes us incompatible.

**Comment by JoshuaKGoldberg** [comment]
Oh, interesting - how should we change the failing spec tests in #5384 then?

**Comment by bradzacher** [comment]
@JoshuaKGoldberg [this file defines the set of "ignored" fixtures](https://github.com/typescript-eslint/typescript-eslint/blob/dc5419674ed1b8d2dceef40bb061f7a84cacc866/packages/typescript-estree/tests/ast-alignment/fixtures-to-test.ts) - we can just add the failing fixtures.

Ideally we should prioritise migrating all the tests across to the `ast-spec` package so that they can just be automatically listed in our spec differences:
- [the global list of misalignments](https://github.com/typescript-eslint/typescript-eslint/blob/dc5419674ed1b8d2dceef40bb061f7a84cacc866/packages/ast-spec/tests/fixtures-with-differences-ast.shot)
- [an individual ast difference snapshot](https://github.com/typescript-eslint/typescript-eslint/blob/dc5419674ed1b8d2dceef40bb061f7a84cacc866/packages/ast-spec/src/declaration/ClassDeclaration/fixtures/implements-many/snapshots/5-AST-Alignment-AST.shot)

**Comment by JoshuaKGoldberg** [comment]
#5384 was merged into `v6` and will be available as part of the v6 release. 🚀
